Sources/JavaScriptKit/BasicObjects/JSPromise.swift

```diff
@@ -40,73 +43,10 @@ public final class JSPromise<Success, Failure>: ConvertibleToJSValue, Constructi
-    */
-    public func then(success: @escaping () -> ()) {
-        let closure = JSOneshotClosure { _ in
-            success()
-            return .undefined
-        }
-        _ = jsObject.then!(closure)
-    }
-
-    /** Schedules the `failure` closure to be invoked on either successful or rejected completion of 
-    `self`.
-    */
-    public func finally(successOrFailure: @escaping () -> ()) -> Self {
-        let closure = JSOneshotClosure { _ in
-            successOrFailure()
-            return .undefined
-        }
-        return .init(unsafe: jsObject.finally!(closure).object!)
-    }
-}
-
-extension JSPromise where Success == (), Failure == Never {
-    /** Creates a new `JSPromise` instance from a given `resolver` closure. `resolver` takes 
-    a closure that your code should call to resolve this `JSPromise` instance.
-    */
-    public convenience init(resolver: @escaping (@escaping () -> ()) -> ()) {
-        let closure = JSOneshotClosure { arguments in
-            // The arguments are always coming from the `Promise` constructor, so we should be
-            // safe to assume their type here
-            resolver { arguments[0].function!() }
-            return .undefined
-        }
-        self.init(unsafe: JSObject.global.Promise.function!.new(closure))
-    }
-}
-
-extension JSPromise where Failure: ConvertibleToJSValue {
-    /** Creates a new `JSPromise` instance from a given `resolver` closure. `resolver` takes 
+    /** Creates a new `JSPromise` instance from a given `resolver` closure. `resolver` takes
     two closure that your code should call to either resolve or reject this `JSPromise` instance.
     */
-    public convenience init(resolver: @escaping (@escaping (Result<Success, JSError>) -> ()) -> ()) {
-        let closure = JSOneshotClosure { arguments in
-            // The arguments are always coming from the `Promise` constructor, so we should be
-            // safe to assume their type here
-            let resolve = arguments[0].function!
-            let reject = arguments[1].function!
-
-            resolver {
-                switch $0 {
-                case .success:
-                    resolve()
-                case let .failure(error):
-                    reject(error.jsValue())
-                }
-            }
-            return .undefined
-        }
-        self.init(unsafe: JSObject.global.Promise.function!.new(closure))
-    }
-}
-
-extension JSPromise where Success: ConvertibleToJSValue, Failure: JSError {
-    /** Creates a new `JSPromise` instance from a given `resolver` closure. `resolver` takes 
-    a closure that your code should call to either resolve or reject this `JSPromise` instance.
-    */
-    public convenience init(resolver: @escaping (@escaping (Result<Success, JSError>) -> ()) -> ()) {
+    public convenience init(resolver: @escaping (@escaping (Result<JSValue, JSValue>) -> ()) -> ()) {
```

Comment:
Could these be `JSValueConvertible` so users can do something like `resolve(.success("Hello, world!"))`?
```suggestion
    public convenience init(resolver: @escaping (@escaping (Result< JSValueConvertible, JSValueConvertible >) -> ()) -> ()) {
```

Reply:
`ConvertibleToJSValue` is not conforming to `Error`, so `Result` can't have Failure as `ConvertibleToJSValue` :cry: